packages/core/src/modules/dids/DidsModuleConfig.ts

```diff
@@ -17,52 +16,79 @@ import {
  */
 export interface DidsModuleConfigOptions {
   /**
-   * List of did registrars that should be registered on the dids module. The registrar must
-   * follow the {@link DidRegistrar} interface, and must be constructable. The registrar will be injected
-   * into the `DidRegistrarService` and should be decorated with the `@injectable` decorator.
+   * List of did registrars that should be used by the dids module. The registrar must
+   * be an instance of the {@link DidRegistrar} interface.
    *
    * If no registrars are provided, the default registrars will be used. The `PeerDidRegistrar` will ALWAYS be
    * registered, as it is needed for the connections and out of band module to function. Other did methods can be
    * disabled.
    *
    * @default [KeyDidRegistrar, IndySdkSovDidRegistrar, PeerDidRegistrar]
    */
-  registrars?: Constructor<DidRegistrar>[]
+  registrars?: DidRegistrar[]
 
   /**
-   * List of did resolvers that should be registered on the dids module. The resolver must
-   * follow the {@link DidResolver} interface, and must be constructable. The resolver will be injected
-   * into the `DidResolverService` and should be decorated with the `@injectable` decorator.
+   * List of did resolvers that should be used by the dids module. The resolver must
+   * be an instance of the {@link DidResolver} interface.
    *
    * If no resolvers are provided, the default resolvers will be used. The `PeerDidResolver` will ALWAYS be
    * registered, as it is needed for the connections and out of band module to function. Other did methods can be
    * disabled.
    *
    * @default [IndySdkSovDidResolver, WebDidResolver, KeyDidResolver, PeerDidResolver]
    */
-  resolvers?: Constructor<DidResolver>[]
+  resolvers?: DidResolver[]
 }
 
 export class DidsModuleConfig {
   private options: DidsModuleConfigOptions
+  private _registrars: DidRegistrar[] | undefined
+  private _resolvers: DidResolver[] | undefined
 
   public constructor(options?: DidsModuleConfigOptions) {
     this.options = options ?? {}
   }
 
   /** See {@link DidsModuleConfigOptions.registrars} */
   public get registrars() {
-    const registrars = this.options.registrars ?? [KeyDidRegistrar, IndySdkSovDidRegistrar, PeerDidRegistrar]
+    // This prevents creating new instances every time this property is accessed
+    if (this._registrars) return this._registrars
 
-    // If the peer did registrar is not included yet, add it
-    return registrars.includes(PeerDidRegistrar) ? registrars : [...registrars, PeerDidRegistrar]
+    let registrars = this.options.registrars ?? [
+      new KeyDidRegistrar(),
+      new IndySdkSovDidRegistrar(),
+      new PeerDidRegistrar(),
+    ]
+
+    // Add peer did registrar if it is not included yet
+    if (!registrars.find((registrar) => registrar instanceof PeerDidRegistrar)) {
```

Comment:
Yes good point. I'll look at a way to do it using the supportedMethods array. Reason I specifically required this one is that the interface for each registrar is different and so a different did:peer registrat may not work. Resolver should be less of an issue. The framework is currently written to work with this specific resolver/registrat

Reply:
Looked at it a bit. Don't think we can easily do this now as we depend on this specific did peer implementation. Any thoughts?

Reply:
I think it's fine if we depend on it right now, as I don't see yet any specific need where we would like to override it. 

We just need make sure that framework will always use it even if user attempts to register another one for did:peer. I think it is the case, because we use Array.find method, which will return the first one added. From my POV there is no need to do a specific check or throwing an specific error as long as we document this constraint somewhere (another TODO for aries-javascript-docs :sweat_smile: )

Reply:
Ah I get what you're saying. Yes good point. We should maybe do an extra check in that case... OR we should only allow a single registrar/resolver to be registered for each method maybe?

Reply:
Well I think it would be better, as you say, to allow a single registrar/resolver per method and throw an error if user attempts to register a second one. But that could be part of a separate PR after 0.3.0 release